src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -166,18 +166,13 @@ public static LocalDateTime combineDateTime(String inputDate, String inputTimeHo
             return null;
         }
 
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("EEE, dd MMM, yyyy H.mm");
         String dateTimeString;
         if ("24".equals(inputTimeHours)) {
             dateTimeString = inputDate + " 23.59";
         } else {
             dateTimeString = inputDate + " " + inputTimeHours + ".00";
         }
-        try {
-            return LocalDateTime.parse(dateTimeString, formatter);
-        } catch (DateTimeParseException e) {
-            return null;
-        }
+        return parseLocalDateTime(dateTimeString, "EEE, dd MMM, yyyy H.mm");
```

Comment:
As this may throw an error, we need to validate input before passing it to `combineDateTime`.

Reply:
heh what do you mean? We are inside `combineDateTime`. The try catch block from here is moved to `parseLocalDateTime`

Reply:
Previously, we didn't throw an assertion error.

Reply:
I see. Then shall I just remove the assertion from `parseLocalDateTime`? In the other place the input is also from user data.